### PR TITLE
[MFTF] Refactoring of AdminMassUpdateProductAttributesMissingRequiredFieldTest

### LINKED ISSUE #31160: [Issue] [MFTF] Refactoring of AdminMassUpdateProductAttributesMissingRequiredFieldTest
This issue is automatically created based on existing pull request: magento/magento2#31079: [MFTF] Refactoring of AdminMassUpdateProductAttributesMissingRequiredFieldTest

---------
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Create 2 simple products
2. Go to Admin->Catalog->Products
3. Request mass update for product name
4. Do not fill the "Product Name"
5. Check if the "Product Name" field is required

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Create 2 simple products
2. Go to Admin->Catalog->Products
3. Request mass update for product name
4. Do not fill the "Product Name"
5. Check if the "Product Name" field is required

### Resolved issues:
1. [x] resolves magento/magento2#31160: [MFTF] Refactoring of AdminMassUpdateProductAttributesMissingRequiredFieldTest

### CODE REVIEW

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminClickSaveOnProductsMassAttributesUpdateActionGroup">
+        <annotations>
+            <description>Clicks on 'Save' button on products mass attributes update page.</description>
+        </annotations>
+
+        <click selector="{{AdminEditProductAttributesSection.Save}}" stepKey="save"/>
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml:16]: If you save something, is much better to make sure the page is loaded afterwards `waitForPageLoad`, including a success/error message if that's the case.
Also please remove the empty rows in your changed files.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    </actionGroup>
+</actionGroups>
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml]: Preserve an empty row at the end of the file.
```suggestion
</actionGroups>

```

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckProductOnProductGridActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            <description>Check the checkbox for the product on the Product Grid</description>	
+        </annotations>	
+        <arguments>	
+            <argument name="product" type="entity"/>	
+        </arguments>	
+
+      <checkOption selector="{{AdminProductGridSection.productRowCheckboxBySku(product.sku)}}" stepKey="selectProduct"/>    	
+
+    </actionGroup>	
+</actionGroups>
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckProductOnProductGridActionGroup.xml]:
```suggestion
</actionGroups> 

```

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckNameToggleOnProductsMassAttributeUpdateActionGroup.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminCheckNameToggleOnProductsMassAttributesUpdateActionGroup">
+        <annotations>
+            <description>Click the "Change" checkbox for the "Name" field on the Products Masss Attributes Update page.</description>
+        </annotations>
+         
+        <click selector="{{AdminEditProductAttributesSection.ChangeAttributeNameToggle}}" stepKey="toggleToChangeName"/>
+
+    </actionGroup>
+</actionGroups>
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckNameToggleOnProductsMassAttributeUpdateActionGroup.xml]:
```suggestion
</actionGroups>

```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @AnnaAPak, thank you for the improvement, could you please check the below minor suggestions?
Thanks.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml (hunk 1 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml:16]: @eduard13 thank you for the advice - this has been addressed.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml (hunk 2 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminClickSaveOnProductsMassAttributeUpdateActionGroup.xml]: Empty row has been added, thanks.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckProductOnProductGridActionGroup.xml (hunk 3 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckProductOnProductGridActionGroup.xml]: Empty row has been added, thanks.

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckNameToggleOnProductsMassAttributeUpdateActionGroup.xml (hunk 4 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminCheckNameToggleOnProductsMassAttributeUpdateActionGroup.xml]: Empty row has been added, thanks.

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run all tests

**Review by eduard13 — APPROVED** [review]
Looks 👍
The failed test doesn't seem to be related to these changes.

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8511](https://jira.corp.magento.com/browse/ENGCOM-8511) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
@magento create issue

**Comment by engcom-Delta** [comment]
**Note:** QA not applicable

**Comment by engcom-Delta** [comment]
**Note:** Functional Tests B2B are failed

**Pull request merged**